### PR TITLE
Expanding Numerical Accuracy Test Operator

### PR DESCRIPTION
**Pull request by jhou-jpg**
Summary:
This diff introduces new parser flag --bitwise and --determinism-check

--bitwise flag can be used to enforce strict tolerance
--determinism-check flag can be use to turn a test into a determinism test, running a set number of times to ensure output are bitwise accurate.

this diff expand operator accuracy regression test coverage, introducing tests for:

### **Attention Operators**
1.  **`flash_attention_accuracy`**

    *   **Baseline**: `aten`
    *   **Test implementations**: `sdpa`, `triton_tutorial_flash_v2`
2.  **`rms_norm_accuracy`**

    *   **Baseline**: `eager_gdpa`
    *   **Test implementations**: `gdpa`, `gdpa_opt`, `gdpa_opt_sorted`, `torch_compile_gdpa`, `torch_compile_rms`

### **Normalization Operators**

1.  **`layer_norm_accuracy`**

    *   **Baseline**: `torch_layer_norm`
    *   **Test implementations**: `triton_layer_norm`, `triton_fused_layer_norm`, `torch_compile_layer_norm`
2.  **`rms_norm_accuracy`**

    *   **Baseline**: `llama_rms`
    *   **Test implementations**: `triton_fused_rmsnorm`, `torch_compile_rms`
3.  **`welford_accuracy`**

    *   **Baseline**: `eager_layer_norm`
    *   **Test implementations**: `triton_welford`, `test_no_welford`, `torch_compile_layer_norm`

### **Softmax Operators**

4.  **`softmax_accuracy`**

    *   **Baseline**: `naive_softmax`
    *   **Test implementations**: `triton_softmax`, `torch_compile_softmax`
5.  **`jagged_softmax_accuracy`**

    *   **Baseline**: `torch_jagged_softmax_unbind_torch_softmax` (torch unbind + softmax)
    *   **Test implementations**: `torch_jagged_softmax_torch_sum`, `triton_jagged_softmax_simple_fused`, `triton_jagged_softmax_variable_length_loop`

### **Reduction Operators**

6.  **`sum_accuracy`**

    *   **Baseline**: `torch_sum` (torch.sum)
    *   **Test implementations**: `triton_sum`, `torch_compile_sum`
7.  **`jagged_sum_accuracy`**

    *   **Baseline**: `torch_jagged_sum_no_pad` (torch sum without padding)
    *   **Test implementations**: `triton_jagged_sum_no_pad_simple_fused`
8. **`jagged_mean_accuracy`**

    *   **Baseline**: `torch_jagged_mean_torch_sum` (torch sum-based mean)
    *   **Test implementations**: `triton_jagged_mean_simple_fused`

Reviewed By: adamomainz, dhruvak3

Differential Revision: D85455951

### CODE REVIEW

**Comment by xuzhao9** [comment]
Please fix the linting error by running `ufmt format .` and address the internal comments, thanks!
